blockbuster/lanes/auction/lane.go

```diff
@@ -47,19 +47,18 @@ func NewTOBLane(
 
 	return &TOBLane{
 		Mempool:     NewMempool(cfg.TxEncoder, maxTx, af),
-		DefaultLane: base.NewDefaultLane(cfg),
+		DefaultLane: base.NewDefaultLane(cfg).WithName(LaneName),
 		Factory:     af,
 	}
 }
 
 // Match returns true if the transaction is a bid transaction. This is determined
 // by the AuctionFactory.
-func (l *TOBLane) Match(tx sdk.Tx) bool {
+func (l *TOBLane) Match(ctx sdk.Context, tx sdk.Tx) bool {
```

Comment:
By having the ctx here do we run into a similar issue where a tx in ProcessProposal is matched against a ctx containing a diff. state that when the tx was inserted into this lane? I.e the same issue we faced when attempting to validate order of tx-priorities in the default-lane

Reply:
Matching doesn't do much in terms of block building. the reason why i think it would be nice to not have it be stateless is that you can extend and have a more robust interface for determining what txs belong to the lane. ultimately, this is meant to be a dev kit and its up to developers to ensure the design makes sense.

That being said, I think its going to be critical for us to set up the best guardrails to ensure developers are doing something unintended. I think docs will be really important moving forward.